### PR TITLE
completion: Don't show one-time switches already in use

### LINKED ISSUE #56: Remove used options from tab completion
```
$ flatpak remote-list --user --user --user -<tab><tab>
-d                --show-details    --show-disabled   --system          --user            -v                --verbose
```

### PR DESCRIPTION
**Pull request by smspillaz**
Fixes #56

### CODE REVIEW

File: common/flatpak-utils.c [hunk 1]
```diff
@@ -3122,6 +3122,29 @@ flatpak_complete_word (FlatpakCompletion *completion,
...
 
+static gboolean
+switch_already_in_line (FlatpakCompletion *completion,
+                        GOptionEntry      *entry)
+{
+  guint i = 0;
+  guint line_part_len = 0;
+  g_auto(GStrv) line = NULL;
+
+  line = g_strsplit (completion->line, " ", -1);
```
Comment (alexlarsson) [inline common/flatpak-utils.c]: Look at completion->argv[] instead, which is pre-split (and shell-unescaped, etc)

File: common/flatpak-utils.c [hunk 2]
```diff
@@ -3122,6 +3122,29 @@ flatpak_complete_word (FlatpakCompletion *completion,
...
+
+  for (; line[i] != NULL; ++i)
+    {
+      line_part_len = strlen (line[i]);
+      if (line_part_len > 2 && g_strcmp0 (&line[i][2], entry->long_name) == 0)
+        return TRUE;
+
+      if (line_part_len == 2 && line[i][1] == entry->short_name)
+        return TRUE;
+    }
```
Comment (alexlarsson) [inline common/flatpak-utils.c]: Certain arguments are useful several times. You need to look at entry->arg, and only drop the argument for these types:
G_OPTION_ARG_NONE,  G_OPTION_ARG_STRING,  G_OPTION_ARG_INT,  G_OPTION_ARG_FILENAME, G_OPTION_ARG_DOUBLE,  G_OPTION_ARG_INT64.

Comment (smspillaz) [inline common/flatpak-utils.c]: Ah, I figures there would be a set list in this case

File: common/flatpak-utils.c (hunk 1 above)
Comment (smspillaz) [inline common/flatpak-utils.c]: Thanks!

**Comment by smspillaz** [comment]
@alexlarsson rebased and updated.

I'm deep-freeing `original_argv` and shallow free-ing `argv` here since the latter gets modified.

File: common/flatpak-utils.c [hunk 3]
```diff
@@ -3122,6 +3122,46 @@ flatpak_complete_word (FlatpakCompletion *completion,
...
+                        GOptionEntry      *entry)
+{
+  guint i = 0;
+  guint line_part_len = 0;
+
+  for (; i < completion->original_argc; ++i)
+    {
+      line_part_len = strlen (completion->original_argv[i]);
+      if (line_part_len > 2 &&
+          g_strcmp0 (&completion->original_argv[i][2], entry->long_name) == 0)
```
Comment (alexlarsson) [inline common/flatpak-utils.c]: I'm not sure why you look at strlen only. You need to verify that it starts with "--", not just anything that is two chars.
I would recommend replacing the whole strlen check with g_str_has_prefix (completion->original_argv[i], "--"), etc.

Comment (alexlarsson) [inline common/flatpak-utils.c]: Oh, i see, you did that before.

Comment (smspillaz) [inline common/flatpak-utils.c]: Hmm. Actually I like your approach better. There may be some validation going on beforehand, but `g_str_has_prefix` better communicates the intent to the reader. Should do a follow-up PR?

Comment (alexlarsson) [inline common/flatpak-utils.c]: Sure, that would be nice.

**Pull request merged**